staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -134,7 +245,19 @@ func (c *unstructuredConverter) FromUnstructured(u map[string]interface{}, obj i
 			klog.Fatalf("FromUnstructured mismatch\nobj1: %#v\nobj2: %#v", obj, newObj)
 		}
 	}
-	return err
+	if err != nil {
+		return err
+	}
+	if strictErrs != nil {
+		return NewStrictDecodingError(strictErrs)
+	}
```

Comment:
if all the unknown field errors are tracked in fromUnstructuredContext, at this point, we can do

```go
if opts.ReturnUnknownFieldErrors && len(fromUnstructuredContext.unknownFieldErrors) > 0 {
  return NewStrictDecodingError(fromUnstructuredContext.unknownFieldErrors)
}
```